spring-web/src/test/java/org/springframework/web/util/UriComponentsBuilderTests.java

```diff
@@ -49,6 +49,18 @@
  */
 class UriComponentsBuilderTests {
 
+	@Test
+	public void fromUriStringQueryParamContainingBracket() throws URISyntaxException {
+		URI uri = new URI("http://example.com/some/path?query=[fromto]");
+		URI finalUri = uri;
+		assertThatIllegalArgumentException().isThrownBy(() -> UriComponentsBuilder.fromUri(finalUri).build(true));
+		//ok, then encode it
+		uri = UriComponentsBuilder.fromUri(uri).build(false).encode().toUri();
+		//confirm encoded
+		URI checkUri = UriComponentsBuilder.fromUri(uri).build(true).toUri();
+		assertThat(uri.toString()).isEqualTo(checkUri.toString());
+	}
+
```

Comment:
Looks like you accidentally included this, which I recognize as a scenario from an unrelated issue.

Reply:
The purpose of this was to check that even though it threw an exception before it works now and is encoded as it should be. I can remove it though if it's unnecessary...